plugins/ksp/src/org/jetbrains/kotlin/ksp/symbol/impl/utils.kt

```diff
@@ -247,6 +248,16 @@ internal fun FunctionDescriptor.toKSFunctionDeclaration(): KSFunctionDeclaration
     }
 }
 
+internal fun PropertyDescriptor.toKSPropertyDeclaration(): KSPropertyDeclaration {
+    if (this.kind != CallableMemberDescriptor.Kind.DECLARATION) return KSPropertyDeclarationDescriptorImpl.getCached(this)
+    val psi = this.findPsi() ?: return KSPropertyDeclarationDescriptorImpl.getCached(this)
+    return when (psi) {
+        is KtProperty -> KSPropertyDeclarationImpl.getCached(psi)
+        is KtParameter -> KSPropertyDeclarationParameterImpl.getCached(psi)
+        else -> throw IllegalStateException("unexpected psi: ${psi.javaClass}")
```

Comment:
Although this is only used in findOverridee and won't encounter PsiField, maybe worth some comments.